geps/gep-1619.md

```diff
@@ -163,43 +163,299 @@ Java application servers such as Tomcat and Jetty, were the first to standardize
+supported and portable; therefore, this GEP will focus on implementing cookie-based session persistence.
 
-To organize potential Gateway API configuration of cookie functionality, the next section will break down configuration into specific categories:
-- Session Cookie Configuration
-- Session Cookie Strategy
+To determine how the session persistence API should be designed, we must consider the following:
+1. Who is responsible for configuring it?
+2. What existing API object(s) is it augmenting?
+3. What are the practical implications of attaching session persistence to an existing API object?
+4. How are existing implementation APIs designed?
 
-**Session Cookie Configuration**
+**Who is responsible for configuring it?**
 
-Session cookie configuration are the parameters and attributes the dataplane uses to create a cookie for establishing a session.
+Referring to the [Gateway API Security Model](https://gateway-api.sigs.k8s.io/concepts/security-model/#roles-and-personas),
+the target kubernetes role/persona for session persistence, are application developers. In most cases, session
+persistence is not considered an optional feature, but rather a necessary and deliberate component incorporated into the
+application's architecture by the developers themselves. Consequently, application developers should be responsible for
+adjusting the configuration to ensure the functionality of their applications.
 
-A set-cookie request can be broken down into [3 main components](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie):
-- Cookie Name
-- Cookie Value
-- Cookie Attributes
+**What existing API object(s) is it augmenting?**
 
-Only the Cookie Name and Cookie Value are required. All Cookie Attributes are optional. Attributes are simply key=value pairs with the value as optional for some attributes.
+Session persistence augments both the concepts of routes and services.
 
-The set-cookies attributes defined by [rfc6265](https://www.rfc-editor.org/rfc/rfc6265#section-5.2) are:
+A Kubernetes service is an abstract representation of a set of pods that work together to provide a specific
+functionality. Services perform load balancing, routing traffic from a service endpoint to a pod. Therefore, session
+persistence augments the load balancing functionality of the service by consistently directing traffic to the same pod.
+
+On the other hand, session persistence also augments routing. An xRoute, as defined by Gateway API, specifies how
+requests from a Gateway listener to an API object (such as a service) are routed. In routing scenarios involving traffic
+splitting, session persistence augments the routing behavior. When a persistent session is established and traffic
+splitting is configured across services, the persistence to a single backend should be maintained across services. In
+other words, the persistent routing behavior should disregard the traffic splitting configuration and remain consistent.
```

Comment:
does this mean that in the case of the following split, once a client get a cookie corresponding to a backend of foo-v1, it will always be routed to that backend, no matter the split?

```yaml
apiVersion: gateway.networking.k8s.io/v1beta1
kind: HTTPRoute
metadata:
  name: simple-split
spec:
  rules:
  - backendRefs:
    - name: foo-v1
      port: 8080
      weight: 90
    - name: foo-v2
      port: 8080
      weight: 10
```

Reply:
Yes, the cookie will make sure the subsequent requests will get routed to that same backend.

Reply:
Technically - the cookie needs to make sure the request goes to the same pod. 
Typically the pods for v1 will be in foo-v1 - so they also go to the same backend.

However in K8S it is possible to change the selector of a Service - for example have foo-prod with 90 and foo-canary with 10, and dynamically change the selector so foo-prod selects as version or another. In such cases it would still be expected that session will persist to the Pod - otherwise things would break. Fortunately this pattern is less common - unfortunately it's more difficult to support with current envoy impl.

Reply:
Good point @costinm. I think you commented this elsewhere. I added this as a use case, not sure if that's the right place, but I felt it should be written down somewhere: https://github.com/kubernetes-sigs/gateway-api/compare/d49134b9d769ce567fa9b3c5181fec3571aa05a3..3e631a6e46895dee15ed90443fbc9647a889d747